### DISCUSSION TITLE
Should `pgxmock.Rows` implement `pgx.Rows` interface?

### DISCUSSION THREAD

**Discussion by VictorDotZ** [pashagolub/pgxmock#119] [Q&A]
`pgx.Rows` returned from `pgxmock.NewRows` is not `pgx.Rows` and can not be passed into a function that handles `pgx.Rows`. But the query with `WillReturnRows` has rows as `pgxmock.rowSets` and all is fine.

So, should `pgxmock.Rows` implement `pgx.Rows` interface, or have you any idea how to test that kind of function without wrapping it by empty queries?

**Comment by pashagolub** [comment]
Hi, would you please provide a code sample that should work and doesn't right now?

Thanks in advance

**Reply by VictorDotZ** [reply]
for example, I have a method that handle `type Rows interface`
```go
func collectPostsFromRows(rows pgx.Rows) (Posts, error) {
	for rows.Next() {
		...
	}
}
``` 
and I want to test this method by passing some rows directly. `pgxmock` provide `NewRows` that passing into `WillReturnRows`, so I try do something like this
```go
rows := pgxmock.NewRows([]string{"field_one", "field_two"}).
	AddRow("some value", "other value").
	AddRow("another value", "something")

collectPostsFromRows(rows)
```

but that's impossible because 
> cannot use rows (variable of type *pgxmock.Rows) as pgx.Rows value in argument to collectPostsFromRows: *pgxmock.Rows does not implement pgx.Rows 

https://github.com/pashagolub/pgxmock/blob/64ed29dc28cc33a74b63031bcc7f7a1d02310094/rows.go#L180

and I have to write something like
```go
mock.ExpectQuery("").WillReturnRows(rows)

returnedRows, _ := mock.Query(context.Background(), "")

collectPostsFromRows(returnedRows)
```
because `WillReturnRows` make `rowSets` from `Rows`, which in turn implement the required interface
https://github.com/pashagolub/pgxmock/blob/64ed29dc28cc33a74b63031bcc7f7a1d02310094/expectations.go#L373
but `rowSets` inaccessible from my code since tha's start with lowercase
https://github.com/pashagolub/pgxmock/blob/64ed29dc28cc33a74b63031bcc7f7a1d02310094/rows.go#L24

that's no critical, but it would be nice to have easy accessible struct for that example

**Reply by pashagolub** [reply]
Sounds good to me. Please, consider providing a pull request for this feature. I'm in a tight loop at the moment
